.github/workflows/integrate.yaml

```diff
@@ -24,64 +24,47 @@ jobs:
     - name: Check flake8
       run: flake8
 
-  test:
-    name: Test
+  deploy:
+    name: Integration Test
     runs-on: ubuntu-latest
 
     steps:
     - name: Check out repo
       uses: actions/checkout@v2
 
-    - uses: balchua/microk8s-actions@v0.2.2
+      # TODO update with main actions-operator once Aggelos PR is merged
+    - name: Setup operator environment
+      uses: neoaggelos/actions-operator@attempt-fix-microk8s-1.19-issue
       with:
-        addons: '["dns", "storage", "rbac"]'
+          provider: microk8s
+          channel: 1.19
 
-    - name: Install dependencies
+    - name: Install Latest Charmcraft
       run: |
         set -eux
-        sudo pip3 install charmcraft
-        sudo snap install juju --classic
-        sudo snap install juju-wait --classic
-        sudo snap install yq
+        sudo snap install charmcraft --classic
+        sudo snap refresh charmcraft --channel=latest/candidate
 
-    - name: Install KNative
+    - name: Enable Microk8s AddOn Knative 
       run: |
         set -eux
-        sudo microk8s enable knative
+        sg microk8s -c "microk8s enable knative"
 
-    - name: Bootstrap Juju
+    # TODO: Remove once the actions-operator does this automatically
+    - name: Configure kubectl
       run: |
-        set -eux
-        sg microk8s -c 'juju bootstrap microk8s uk8s'
-        juju add-model ci-test
+        sg microk8s -c "microk8s config > ~/.kube/config"
 
-    - name: Deploy charm
+    - name: Deploy and Test Charm
       run: |
-        set -eux
-        charmcraft build
-        juju deploy ./*.charm \
-          --resource oci-image=$(yq eval '.resources.oci-image.upstream-source' metadata.yaml)
-        juju wait -wvt 300
-
-    - name: Test charm
-      run: |
-        set -eux
-        kubectl apply -nci-test -f examples/flower.yaml
-        sleep 10
-        kubectl wait -nci-test --for=condition=available deployment --all --timeout=5m
-        ADDRESS=$(kubectl get pods -nci-test -lcomponent=predictor -oyaml | yq e .items.0.status.podIP -)
-
-        curl -v \
-          -d @examples/flower_input.json \
-          -H "Content-Type: application/json" \
-          http://$ADDRESS:8080/v1/models/my-model:predict
+        sg microk8s -c "tox -e integration -- --model testing"
```

Comment:
I don't think the `sg` wrapper call here is necessary. The tests don't use `microk8s kubectl` at all currently. If they did eventually need to use `kubectl`, the bit where you set up `~/.kube/config` should be enough to avoid having to use `microk8s kubectl` directly.